[Emberlynn Loo] iP

Emberlynn-Loo opened the pull request:
**TalkToMeOrIllDie**

“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

DukePro frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ _SUPER_ FAST to use

All you need to do is,

1. download it from [here](https://nus-cs2103-ay2324s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

- [x]  Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)
______________
If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```
public static void main(String[] args) throws DukeException {
    new Duke().run();
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

Commit:
docs/README.md: Tweak document template

Let's tweak the docs/README.md (which is used as the user guide)
to fit Duke better. Specifically,

1. mention product name in the title
2. mention adding a product screenshot and a product intro
3. tweak the flow to describe feature-by-feature

Emberlynn-Loo committed:
Rename, Greet, Exit

Emberlynn-Loo committed:
Echo

Emberlynn-Loo committed:
Add, List

Emberlynn-Loo committed:
Mark as Done

Emberlynn-Loo committed:
Mark as Done

Emberlynn-Loo committed:
ToDo, Event, Deadline

Emberlynn-Loo committed:
Handle Errors

Emberlynn-Loo committed:
Delete

Emberlynn-Loo committed:
Saving it to the Hardware P1

Emberlynn-Loo committed:
Saving it to the Hardware P2

Emberlynn-Loo committed:
Saving it to the Hardware Finished with exceptions

Emberlynn-Loo committed:
Date and Time

whelan-low commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+
+    @Override
+    public String toSaveString() {
+        return "D | " + (isDone ? "1" : "0") + " | " + description + " | " + by.format(DateTimeFormatter.ofPattern("yyyy-MM-dd HHmm"));
```
A line break here is necessary, the current line length is above 120 characters

whelan-low commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,23 @@
+import java.time.format.DateTimeFormatter;
+
+public class Deadline extends Task {
+    private final LocalDateTime by;
```
Perhaps a more descriptive name would be better, such as 'byDate' or 'byTime'

whelan-low commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,275 @@
+                }
+                System.out.println("    ____________________________________________________________");
+            } else if (echo.startsWith("mark") || echo.startsWith("Mark")) {
+                int pos = echo.indexOf(" ");
```
position might be a better variable name than pos

whelan-low reviewed (commented):
Overall, good job on catching both upper and lower case for the key words. Mainly work on giving more descriptive names for the variables.

Emberlynn-Loo committed:
Used More OOP

Emberlynn-Loo committed:
Organized into Packages

Emberlynn-Loo committed:
Merge commit 'a6f73244a20539e20c2189973ebf597df34ef631'

Emberlynn-Loo committed:
Use Gradle

dedsecrattle commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,37 @@
+package Duke;
+
+import java.io.*;
```
Should this wildcard import be replaced with specific imports required by program?

Emberlynn-Loo committed:
Added JUnit tests

Emberlynn-Loo committed:
Created a JAR file

Emberlynn-Loo committed:
Added a Find function

Emberlynn-Loo committed:
Handled some exceptions

Emberlynn-Loo committed:
Revert "Handled some exceptions"

This reverts commit d3787c36962d50684522bd0454913c8479a816a6.

Emberlynn-Loo committed:
Added find function

Emberlynn-Loo committed:
Revert "Added find function"

This reverts commit 1a327d33feb9027c4e29a49b756162c28a65c4d9.

Emberlynn-Loo committed:
Added find fucntion

Emberlynn-Loo committed:
Merge branch 'master' of https://github.com/Emberlynn-Loo/ip

Emberlynn-Loo committed:
Merge branch 'branch-Level-9'

dedsecrattle commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,140 @@
+
+    public static String parse(String userInput, TaskList tasks, Ui ui, Storage storage) throws DukeException {
+        try {
+            if (userInput.equals("bye") || userInput.equals("Bye")) {
```
Any reason why you did it using `if-else` instead of using `switch-case`?

dedsecrattle commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,106 @@
+    }
+
+
+    public void showTasks(Task[] task,int taskNum){
```
Curly braces `{` can be proceeded by a whitespace ?

jxunze commented on src/Test/java/Duke/DeadlineTest.java:
```diff
@@ -0,0 +1,21 @@
+public class DeadlineTest {
+
+    @Test
+    public void testGetBy() {
```
Do consider looking into junit naming convention for the tests. Test name should express a specific requirement. Here is a [link](https://medium.com/@stefanovskyi/unit-test-naming-conventions-dd9208eadbea) for some of the naming conventions.

jxunze commented on src/Test/java/Duke/DukeExceptionTest.java:
```diff
@@ -0,0 +1,12 @@
+public class DukeExceptionTest {
+
+    @Test
+    public void testDukeExceptionMessage() {
```
ditto

jxunze commented on src/Test/java/Duke/DukeTest.java:
```diff
@@ -0,0 +1,11 @@
+
+public class DukeTest {
+    @Test
+    public void dukeConstructorTest() {
```
ditto

jxunze commented on src/Test/java/Duke/EventTest.java:
```diff
@@ -0,0 +1,24 @@
+
+public class EventTest {
+    @Test
+    public void testEventToString() {
```
ditto

jxunze commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,140 @@
+
+public class Parser {
+
+    public static String parse(String userInput, TaskList tasks, Ui ui, Storage storage) throws DukeException {
```
This function is very long. Consider breaking it up into separate functions and adding comments. It will help with readability.

jxunze commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,140 @@
+                storage.saveTasksToFile(TaskList.getTasks(), TaskList.getTaskNum());
+                ui.goodbyeMessage();
+                return "1";
+            } else if (userInput.equals("list") || userInput.equals("List")) {
```
I think using `userInput.toLowerCase()` will make your code cleaner.

jxunze commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,140 @@
+                        throw new DukeException("Invalid task number >:((");
+                    }
+                }
+            } else if (userInput.startsWith("todo") || userInput.startsWith("Duke.Todo")) {
```
Is there a reason why you check for "todo" and "Duke.Todo"?

jxunze commented on src/main/java/Duke/Storage.java:
```diff
@@ -0,0 +1,100 @@
+package Duke;
```
Consider adding javadocs to this file!

jxunze reviewed (commented):
Overrall, only minor errors with regards to coding conventions, most of which are the naming for the test functions. Good job!

Emberlynn-Loo committed:
Added JavaDoc comments

Emberlynn-Loo committed:
Revert "Added JavaDoc comments"

This reverts commit ea40507a7cb21e9fcbe46eb5e5bad5cbeba9ce47.

Emberlynn-Loo committed:
Revert "Revert "Added JavaDoc comments""

This reverts commit 941dd2e608886dde33aa88051b135dd91b71e9a8.

Emberlynn-Loo committed:
Tweak the code to comply with a coding standard

Emberlynn-Loo committed:
delete unused test

Emberlynn-Loo committed:
Add checkstyle

Emberlynn-Loo committed:
Add GUI to the chatbot

Emberlynn-Loo committed:
Use Java varargs feature

Emberlynn-Loo committed:
Shorten long methods by extracting some code blocks into separate methods and use assert feature

Emberlynn-Loo committed:
Refactor code for improved clarity and adherence to coding standards

UI Class:
Shortened the welcomeMessage() method for brevity and clarity.
Changed variable names to be more descriptive and adhere to code quality standards.
Storage Class:
Renamed variables for clarity, e.g., sep1 to separation1.
Updated method headers (Javadocs) to conform to standard format.
Parser Class:
Made method names more specific to reduce the number of methods and improve clarity.
Duke Class:
Improved variable naming, e.g., userInput to userInputField.
Adjusted header comments to follow the specified format for clarity and consistency.
The changes aim to enhance code readability, maintainability, and adherence to coding standards.

This commit addresses the need for clear and concise code, making it easier to understand and maintain. It improves variable naming, method clarity, and adheres to standard formatting guidelines.

These changes were necessary to ensure the codebase meets quality standards and remains easy to maintain in the future.

Emberlynn-Loo committed:
Add extension to detect duplicate items

Emberlynn-Loo committed:
Add more flexibility in searching for items

When using the find command, the user can find
items even if the keyword matches the item only partially
(eg. find boo will show matching task: "[T][] return book")

Emberlynn-Loo committed:
Merge pull request #2 from Emberlynn-Loo/branch-A-Duplicate

Add extension to detect duplicate items

Emberlynn-Loo committed:
Merge branch 'master' into branch-C-BetterSearch

# Conflicts:
#	data/duke.txt
#	src/main/java/duke/Ui.java

Emberlynn-Loo committed:
Merge pull request #3 from Emberlynn-Loo/branch-C-BetterSearch

Add more flexibility in searching for items

Emberlynn-Loo committed:
Give help to new users

Add functionality to give a list of all available commands
by typing help

Tsenrae commented on src/main/resources/images/DaDuke.png:
Wow! What an amazing depiction of an average CS student. This image simply leaves me in awe of its greatness

Emberlynn-Loo committed:
Create new jar file to be able to launch GUI

Emberlynn-Loo committed:
Edit help function to include all commands

Emberlynn-Loo committed:
Fix errors so 'on' command works properly

Emberlynn-Loo committed:
Add files via upload

Emberlynn-Loo committed:
Ensure consistent product name in GUI

Emberlynn-Loo committed:
Add a User Guide

Emberlynn-Loo committed:
Add more to the user guide

Emberlynn-Loo committed:
Final edits

Emberlynn-Loo committed:
Add Ui screenshot

Emberlynn-Loo committed:
Add Ui Screenshot

Emberlynn-Loo committed:
Merge branch 'branch-A-Release'

Emberlynn-Loo committed:
Make Jar file fun on Windows

Emberlynn-Loo committed:
Merge branch 'branch-A-Release'
